compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java

```diff
@@ -0,0 +1,606 @@
+    }
+
+    public void visit(BIRTerminator.Flush entry) {
+        writePosition(entry.pos);
+        buf.writeByte(entry.kind.getValue());
+        buf.writeInt(entry.channels.length);
+        for (BIRNode.ChannelDetails detail : entry.channels) {
+            addCpAndWriteString(detail.name);
+            buf.writeBoolean(detail.channelInSameStrand);
+            buf.writeBoolean(detail.send);
+        }
+        entry.lhsOp.accept(this);
+        addCpAndWriteString(entry.thenBB.id.value);
+    }
+
+    public void visit(BIRTerminator.WorkerReceive entry) {
+        writePosition(entry.pos);
+        buf.writeByte((entry.kind.getValue()));
+        buf.writeInt(addStringCPEntry(entry.workerName.getValue()));
+        entry.lhsOp.accept(this);
+        buf.writeBoolean(entry.isSameStrand);
+        addCpAndWriteString(entry.thenBB.id.value);
+    }
+
+    public void visit(BIRTerminator.WorkerSend entry) {
+        writePosition(entry.pos);
+        buf.writeByte((entry.kind.getValue()));
+        buf.writeInt(addStringCPEntry(entry.channel.getValue()));
+        entry.data.accept(this);
+        buf.writeBoolean(entry.isSameStrand);
+        buf.writeBoolean(entry.isSync);
+        if (entry.isSync) {
+            entry.lhsOp.accept(this);
+        }
+        addCpAndWriteString(entry.thenBB.id.value);
+    }
+
+    public void visit(BIRTerminator.WaitAll waitAll) {
+        writePosition(waitAll.pos);
+        buf.writeByte((waitAll.kind.getValue()));
+        waitAll.lhsOp.accept(this);
+        buf.writeInt(waitAll.keys.size());
+        waitAll.keys.forEach(key -> buf.writeInt(addStringCPEntry(key)));
+        waitAll.valueExprs.forEach(val -> val.accept(this));
+        addCpAndWriteString(waitAll.thenBB.id.value);
+    }
+
+    // Non-terminating instructions
+
+    @Override
+    public void visit(BIRNonTerminator.NewTable newTable) {
+        writePosition(newTable.pos);
+        writeType(newTable.type);
+        newTable.lhsOp.accept(this);
+        newTable.keyColOp.accept(this);
+        newTable.dataOp.accept(this);
+    }
+
+
+    public void visit(BIRNonTerminator.Move birMove) {
+        writePosition(birMove.pos);
+        buf.writeByte(birMove.kind.getValue());
+        birMove.rhsOp.accept(this);
+        birMove.lhsOp.accept(this);
+    }
+
+    public void visit(BIRTerminator.Call birCall) {
+        writePosition(birCall.pos);
+        buf.writeByte(birCall.kind.getValue());
+        PackageID calleePkg = birCall.calleePkg;
+        int pkgIndex = addPkgCPEntry(calleePkg);
+        buf.writeBoolean(birCall.isVirtual);
+        buf.writeInt(pkgIndex);
+        buf.writeInt(addStringCPEntry(birCall.name.getValue()));
+        buf.writeInt(birCall.args.size());
+        for (BIROperand arg : birCall.args) {
+            arg.accept(this);
+        }
+        if (birCall.lhsOp != null) {
+            buf.writeByte(1);
+            birCall.lhsOp.accept(this);
+        } else {
+            buf.writeByte(0);
```

Comment:
Since we have `buf.writeByte(1);` and `buf.writeByte(0);` in multiple places wouldn't it be nice to have constants for 1 and 0 indicating what they mean in this context? If that makes sense 🙂

Reply:
Encoding 1 means true and and 0 means false. For me 1 and 0 more readable for me than replace it with constants.

Reply:
I thought for someone like me who doesn't know what 1 and 0 stand for, it would make a lot of sense if it is a constant like TRUE_VAL for 1